src/openzeppelin/tests/test_accesscontrol.cairo

```diff
@@ -2,124 +2,308 @@ use openzeppelin::access::accesscontrol::AccessControl;
-    assert(!AccessControl::has_role(DEFAULT_ADMIN_ROLE, ACCOUNT1()), 'Role should be revoked');
+    AccessControl::revoke_role(ROLE, AUTHORIZED());
+    AccessControl::revoke_role(ROLE, AUTHORIZED());
+    assert(!AccessControl::has_role(ROLE, AUTHORIZED()), 'Role should still be revoked');
 }
 
 #[test]
 #[available_gas(2000000)]
 #[should_panic(expected: ('Caller is missing role', ))]
 fn test_revoke_role_unauthorized() {
     setup();
-    set_caller_address(ACCOUNT2());
-    AccessControl::revoke_role(DEFAULT_ADMIN_ROLE, ACCOUNT1());
+
+    set_caller_address(OTHER());
+    AccessControl::revoke_role(ROLE, AUTHORIZED());
+}
+
+//
+// renounce_role
+//
+
+#[test]
+#[available_gas(2000000)]
+fn test_renounce_role_for_role_not_granted() {
+    setup();
+    set_caller_address(AUTHORIZED());
+
+    AccessControl::renounce_role(ROLE, AUTHORIZED());
+    assert(!AccessControl::has_role(ROLE, AUTHORIZED()), 'Role should be renounced');
 }
 
 #[test]
 #[available_gas(2000000)]
-fn test_renounce_role() {
+fn test_renounce_role_for_granted_role() {
     setup();
-    AccessControl::renounce_role(DEFAULT_ADMIN_ROLE, ACCOUNT1());
-    assert(!AccessControl::has_role(DEFAULT_ADMIN_ROLE, ACCOUNT1()), 'Role should be renounced');
+    AccessControl::grant_role(ROLE, AUTHORIZED());
+    set_caller_address(AUTHORIZED());
+
+    AccessControl::renounce_role(ROLE, AUTHORIZED());
+    assert(!AccessControl::has_role(ROLE, AUTHORIZED()), 'Role should be renounced');
+}
+
+#[test]
+#[available_gas(2000000)]
+fn test_renounce_role_multiple_times_for_granted_role() {
+    setup();
+    AccessControl::grant_role(ROLE, AUTHORIZED());
+    set_caller_address(AUTHORIZED());
+
+    AccessControl::renounce_role(ROLE, AUTHORIZED());
+    AccessControl::renounce_role(ROLE, AUTHORIZED());
+    assert(!AccessControl::has_role(ROLE, AUTHORIZED()), 'Role should still be renounced');
 }
 
 #[test]
 #[available_gas(2000000)]
 #[should_panic(expected: ('Can only renounce role for self', ))]
 fn test_renounce_role_unauthorized() {
     setup();
-    set_caller_address(ACCOUNT2());
-    AccessControl::renounce_role(DEFAULT_ADMIN_ROLE, ACCOUNT1());
+    AccessControl::grant_role(ROLE, AUTHORIZED());
+
+    // Admin is unauthorized caller
+    AccessControl::renounce_role(ROLE, AUTHORIZED());
 }
 
+//
+// _set_role_admin
+//
+
 #[test]
 #[available_gas(2000000)]
-fn test_set_role_admin() {
+fn test__set_role_admin() {
     setup();
-    AccessControl::_set_role_admin(DEFAULT_ADMIN_ROLE, SOME_OTHER_ROLE);
     assert(
-        AccessControl::get_role_admin(DEFAULT_ADMIN_ROLE) == SOME_OTHER_ROLE,
-        'Should set the admin role'
+        AccessControl::get_role_admin(ROLE) == DEFAULT_ADMIN_ROLE, 'ROLE admin default should be 0'
     );
+    AccessControl::_set_role_admin(ROLE, OTHER_ROLE);
+    assert(AccessControl::get_role_admin(ROLE) == OTHER_ROLE, 'ROLE admin should be OTHER_ROLE');
+}
+
+#[test]
+#[available_gas(2000000)]
+fn test_change_an_admin_roles_admin() {
```

Comment:
i don't think i understand this test. what does it test?

Reply:
Hm it's really repeating the initial `test__set_role_admin`. I'll remove